Add SCRIPT_ALLOW_SEGWIT_RECOVERY

sickpig opened the pull request:
Based on top of #1644 
This still need the activation code and python tests. Will add later today

dagurval commented:
Last two commits, _port of 449f9b8_ and _Add SCRIPT_ALLOW_SEGWIT_RECOVERY_ LGTM.

sickpig commented:
@dagurval what's LGTM?

sickpig committed:
Partial port of Core's 449f9b8d (BIP141)

Add a helper to determine if CScript is a witness script

sickpig committed:
Add SCRIPT_ALLOW_SEGWIT_RECOVERY

Summary: Flag that allows the spending of p2sh segwit coins

Test Plan: test_bitcoin --run_test=script_tests

Reviewers: deadalnix, O1 Bitcoin ABC, #bitcoin_abc

Reviewed By: deadalnix, O1 Bitcoin ABC, #bitcoin_abc

Subscribers: markblundeberg, teamcity, schancel

Differential Revision: https://reviews.bitcoinabc.org/D2468

floriansengel committed:
Partial port of "Add activation code for SEGWIT_RECOVERY" (D2479)

Test Plan: test_runner.py abc-segwit-recovery-activation

Reviewers: deadalnix, O1 Bitcoin ABC, #bitcoin_abc, jasonbcox, florian

Reviewed By: deadalnix, O1 Bitcoin ABC, #bitcoin_abc, jasonbcox

Subscribers: Fabien, Mengerian, markblundeberg, jasonbcox, teamcity, schancel

Differential Revision: https://reviews.bitcoinabc.org/D2479
